src/main/java/fr/zcraft/quartzlib/components/nbt/NBTType.java

```diff
@@ -190,17 +243,79 @@ public Object getData(Object nmsNbtTag) {
             return null;
         }
         try {
+            switch (getNmsTagFieldName()) {
+                case "map":
+                    //Since 1.17 "map" became "tags"
+                    //h() return an unmodifiable map
+                    return Reflection.call(nmsNbtTag.getClass(), nmsNbtTag, "h");
+
+                case "list":
+                    //We recreate the list because there are no getter for the list.
+                    //TODO check if in 1.18 a getter is added for this one.
+                    List list = new ArrayList();
+
+                    for (int i = 0; i < (int) Reflection.call(nmsNbtTag.getClass(), nmsNbtTag, "size"); i++) {
+                        //Strange thing of java a call cast int into integer resulting in the impossibility
+                        // to call the method
+                        Method method;
+
+
+                        method = nmsNbtTag.getClass().getMethod("get", int.class);
+
+
+                        list.add(method.invoke(nmsNbtTag, i));
+                        //list.add(Reflection.call(nmsNbtTag.getClass(), nmsNbtTag, "get", i));
+                        // Doesn't work see above
+                    }
+                    return list;
+                case "data":
+                    PluginLogger.info("To be implemented");
+                    //TODO I don't know what to do here
+                    break;
+                default:
+                    break;
+            }
+            //Fallback used for data right now don't know if this works
             return Reflection.getFieldValue(nmsNbtTag, getNmsTagFieldName());
-        } catch (Exception ex) {
-            throw new NBTException("Unable to retrieve NBT tag data", ex);
+        } catch (Exception exc) {
+            //Older versions than 1.17 (fields are no longer accessible in java 17)
+            try {
+                return Reflection.getFieldValue(nmsNbtTag, getNmsTagFieldName());
+            } catch (Exception e) {
+                try {
+                    return Reflection.getFieldValue(nmsNbtTag, getNmsTagFieldName());
+                } catch (Exception ex) {
+                    throw new NBTException("Unable to retrieve NBT tag data", ex);
+                }
+            }
         }
+
+
     }
 
     public void setData(Object nmsNbtTag, Object value) {
         try {
-            Reflection.setFieldValue(nmsNbtTag, getNmsTagFieldName(), value);
-        } catch (Exception ex) {
-            throw new NBTException("Unable to set NBT tag data", ex);
+            switch (getNmsTagFieldName()) {
+                case "map":
+                    Reflection.call(nmsNbtTag.getClass(), nmsNbtTag, "set", getNmsTagFieldName(), value);
+                    break;
+                //TODO check if this works with every type,
+                // if not has to do by hand all calls for set+type and a (UUID),b,c...
```

Comment:
And the answer is kinda no, I leave this right here because there are still some to add. They are not important nor used that much but adding them will be needed in a future update

Reply:
Yup but the next line as to be a TODO this one should stay because we may have to at some point add various call to nbt setters for each type (same for getter, that's a pain in java17) (they are named a,b,c in 1.18)